### PR TITLE
Hylics 2: Fix invalid multiworld data, use `self.random` instead of `self.multiworld.random`

### PR DESCRIPTION
**Pull request by TRPG0**
I should probably just rewrite the whole Hylics 2 world at this point

### CODE REVIEW

**Review by ScipioWright — APPROVED** [review]
Code looked fine, test gens worked fine.
Would recommend also getting the options updated to the newer options API as well in a separate PR. Just a matter of changing stuff like
`self.multiworld.random_start[self.player]` to `self.options.random_start`, and reformatting the option list at the bottom of options.py mostly

File: worlds/hylics2/__init__.py [hunk 1]
```diff
@@ -107,46 +101,44 @@ def pre_fill(self):
...
-                tv = self.multiworld.random.choice(tvs)
-                self.multiworld.get_location(tv[1]["name"], self.player)\
-                    .place_locked_item(self.add_item(gest[1]["name"], gest[1]["classification"], gest[0]))
+            for _ in range(len(gestures)):
+                gest = self.random.choice(gestures)
+                tv = self.random.choice(tvs)
+                self.multiworld.get_location(tv, self.player)\
+                    .place_locked_item(self.create_item(gest))
                 gestures.remove(gest)
                 tvs.remove(tv)
```
Comment (alwaysintreble) [inline worlds/hylics2/__init__.py:143]:
```suggestion
            self.random.shuffle(gestures)
            self.random.shuffle(tvs)
            while gestures:
                gesture = gestures.pop()
                tv = tvs.pop()
                self.get_location(tv).place_locked_item(self.create_item(gesture)
```

Comment (TRPG0) [reply worlds/hylics2/__init__.py:143]: Sure

**Review by silasary — APPROVED** [review]

**Review by silasary — APPROVED** [review]
This is a good refactor

**Comment by black-sliver** [comment]
It is unclear to me what bug this fixes (since it's labelled as bug/fix). Can you elaborate?

**Comment by TRPG0** [comment]
It's fixing this (again)

**Review by black-sliver — APPROVED** [review]
I don't see what changed to make that error disappear, but the code changes look fine anyway and I tested rolling with shuffle_gestures: tvs_only, which was previously broken and works now.

**Comment by black-sliver** [comment]
Oh, I see the error now. You pass the item data into the code field. Yeah, the new code is way less error prone :-)

**Pull request merged**